### ISSUE TITLE
Access Redshift-Independent Distances from NED

### ISSUE THREAD

**Issue by brianhausmann** [astropy/astroquery#606]
Hi, 

I want to retrieve redshift-independent distances from NED for a list of objects. Looking through the documentation on astroquery.ned, it seemed that I should use the get_table() function, although this can only be set to one of five possible keywords, none of which will return a redshift-independent distance value. Does anyone know if perhaps the documentation is incomplete, and there is some other way that I can use astroquery to return these distances from NED?

**Comment by keflavich** [comment]
I'm not very familiar with NED, so I can't answer this directly, but could you say how you would do it via the NED website?  If you could show the HTTP POST data, for example, we can probably tell you how to use astroquery to do it.

**Comment by brianhausmann** [comment]
In order to access these distances via the NED website, I would use this query form: 

http://ned.ipac.caltech.edu/forms/gmd.html 

Here is the page of results for a few example galaxies:

http://ned.ipac.caltech.edu/cgi-bin/gmd?uplist=M31%0D%0AM32%0D%0ANGC+1068%0D%0ANGC+4151&delimiter=bar&nondb=row_count&nondb=user_name_msg&nondb=user_objname&crosid=objname&position=ra%2Cdec&enotes=objnote&position=pretype&position=z&position=zflag&distance_CON=dmpc&attdat=attned&gphotoms=q_value&gphotoms=q_unc&gphotoms=ned_value&gphotoms=ned_unc&diamdat=ned_maj_dia&distance=avg&distance=stddev_samp

Does that help?

**Comment by keflavich** [comment]
Yes, that's helpful, sorry about the delay.  Unfortunately, it looks like we have not yet implemented a NED tool for this type of query, though I can't say for sure.  @willettk, do you have any suggestions about how to do this with existing machinery, or do we need to implement new tools?

**Comment by willettk** [comment]
I would imagine it can be done - NED made a lot of changes to the UI recently, and I'm sure we could incorporate some of them into astroquery. It's not searchable right now, as far as I know. 

Basically we want the results of page:

http://ned.ipac.caltech.edu/cgi-bin/nDistance?name=NGC+1300

to be returned as an object that astroquery can parse. I tried the adding the standard `&of=xml_main` to this page, but it doesn't work. 

I sent a question to the NED helpdesk asking if it's possible.

**Comment by brianhausmann** [comment]
@willetk what you are saying is correct. I have previously contacted the helpdesk, and received the response that the functionality is not currently available. They claimed that they are working on it, and that it should be available in a few months or so.

**Comment by willettk** [comment]
Good to know.

I suppose you could write your own web-scraping query with BeautifulSoup or
something similar if you wanted to deal with the raw HTML page that
currently exists, but I think we should wait to update astroquery until NED
has an easily exportable format.

**Comment by brianhausmann** [comment]
I have managed to successfully write my own web-scraping query, I was only interested in exhausting all other options in the meantime. I appreciate the assistance.

**Comment by willettk** [comment]
My response from NED, for posterity: 

"Thank you for asking about a way of obtaining a flat-file version of the
data contained in the redshift-independent database we call NED-D.

In case you weren't aware, the NED home page has a selection called
"Galaxy Distance Tabulations (NED-D)" within which you can download
a .csv version of that dataset. Be aware that this file is updated
with every new NED release (~2-3 months).

As for having the NED interface output these data in XML or another
format, this is part of a major change to our database we are currently
working on. User-specified output data types are a big part of this
upgrade."

**Comment by keflavich** [comment]
@bsipocz I think we can close this unless you know of improvements being made in NED to enable this sort of problem?  The original user's issue is solved.
